backstory/src/main/java/com/google/sps/story/PromptManager.java

```diff
@@ -13,46 +13,77 @@
 // limitations under the License.
 package com.google.sps.story;
 
+import com.google.sps.story.data.*;
+import java.lang.IllegalArgumentException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Scanner;
 
 /**
  * Creates prompt string for text generation using input keyword strings.
  */
 public final class PromptManager {
+  /** Keywords for generation */
   private List<String> keywords;
+  /** Word fetching/processing client */
+  private PromptManagerWordTools wordTools;
+  /** Randomness flag. */
+  private boolean chooseRandomly;
 
   /**
-   * Initialize labels parameter.
+   * Initialize keywords and randomness parameters.
    *
-   * @param labels A list of Strings containing keywords for prompts.
+   * @param keywords A list of Strings containing keywords for prompts.
    */
   public PromptManager(List<String> keywords) {
     this.keywords = keywords;
+    this.chooseRandomly = true;
   }
 
   /**
-   * Generates prompt using labels.
+   * Sets wordTools for word processing API calls.
+   *
+   * @param wordTools PromptManagerWordTools instance.
+   */
+  public void setWordTools(PromptManagerWordTools wordTools) {
+    this.wordTools = wordTools;
+  }
+
+  /**
+   * Sets use of randomness in prompt template selection.
+   *
+   * @param boolean Whether or not to randomly choose output templates..
+   */
+  public void setRandom(boolean chooseRandomly) {
+    this.chooseRandomly = chooseRandomly;
+  }
+
+  /**
+   * Generates prompt using keywords given.
    *
-   * @param delimiter String for delimiter between appended strings.
    * @return A String containing the output prompt.
    */
-  public String generatePrompt(String delimiter) {
-    String prompt = "";
+  public String generatePrompt() {
+    // Prepare story-like prefix.
+    String prompt = "Once upon a time, ";
 
-    // Check if valid.
-    if (keywords.size() == 0) {
-      return "";
+    // Check for null input.
+    if (keywords == null) {
```

Comment:
Seems like validation should be done in the constructor instead? That way the client would know earlier that this PromptManager instance will never be able to generate a prompt.

Reply:
Done